libyul/optimiser/StructuralSimplifier.cpp

```diff
@@ -16,106 +16,122 @@
 */
 #include <libyul/optimiser/StructuralSimplifier.h>
 #include <libyul/optimiser/Semantics.h>
+#include <libyul/optimiser/Utilities.h>
 #include <libyul/AsmData.h>
 #include <libdevcore/CommonData.h>
+#include <libdevcore/Visitor.h>
 
 using namespace std;
 using namespace dev;
 using namespace yul;
 
+namespace {
+
+ExpressionStatement makePopExpressionStatement(langutil::SourceLocation const& _location, Expression _expression)
```

Comment:
`Expression&&`?

Reply:
I'll argue about that with @christianparpart :-). I think both would be valid and they both have pro's and con's, but in the end both should in fact move (given the correct way to move during the call).